lib/SIL/IR/OperandOwnership.cpp

```diff
@@ -261,6 +265,7 @@ OPERAND_OWNERSHIP(ForwardingBorrow, OpenExistentialValue)
 OPERAND_OWNERSHIP(ForwardingBorrow, OpenExistentialBoxValue)
 
 OPERAND_OWNERSHIP(EndBorrow, EndBorrow)
+OPERAND_OWNERSHIP(EndBorrow, EndApply)
```

Comment:
I don't think this is correct. The end_apply just takes in the token from the begin_apply. What ownership semantics/type does a token have?

Reply:
The token only has ownership semantics. The whole reason it exists is to establish the scope of borrowed coroutine arguments. If it were considered a trivial use then it would be incorrectly ignored by any code that looks for uses relevant to the lifetime of borrowed arguments.

Reply:
Andy and I spoke about this offline. Can you add end_abort here as well and add what we talked about last night to this. I say something like this:

```
NOTE: Remembering that end_apply, abort_apply take a begin_apply token which has None ownership one may wonder why we would mark end_apply, abort_apply as an EndBorrow use. The reason why is that when one is walking def->use for ownership purposes, we will visit the end_apply, abort_apply use as a use of any values that we instantaneously borrow for the coroutine. In such a case, if we marked the use as trivial, we would lose semantic information.
```

I think that would clear up any questions this piece of code brings up.

Reply:
Explained in this commit
https://github.com/apple/swift/pull/35248/commits/07f5735e53bd340dc57ec16287ecc8d24f2b4ad3